xwayland/selection.c

```diff
@@ -55,6 +55,15 @@ static int xwm_read_data_source(int fd, uint32_t mask, void *data) {
 	int current = selection->source_data.size;
 	if (selection->source_data.size < incr_chunk_size) {
 		p = wl_array_add(&selection->source_data, incr_chunk_size);
+		if (!p){
```

Comment:
Now I understand this slightly better, I guess it's probably safer to do the same as read fail (clear x11 selection, remove source and close fd); will do that instead.